contrib/babelfishpg_common/sql/upgrades/babelfish_common_helper--4.3.0--4.4.0.sql

```diff
@@ -0,0 +1,79 @@
+------------------------------------------------------------------------------
+---- Include changes related to other datatypes except spatial types here ----
+------------------------------------------------------------------------------
+
+-- complain if script is sourced in psql, rather than via ALTER EXTENSION
+\echo Use "ALTER EXTENSION ""babelfishpg_common"" UPDATE TO "4.4.0"" to load this file. \quit
+
+SELECT set_config('search_path', 'sys, '||current_setting('search_path'), false);
+
+-- Drops an object if it does not have any dependent objects.
+-- Is a temporary procedure for use by the upgrade script. Will be dropped at the end of the upgrade.
+-- Please have this be one of the first statements executed in this upgrade script. 
+CREATE OR REPLACE PROCEDURE babelfish_drop_deprecated_object(object_type varchar, schema_name varchar, object_name varchar) AS
+$$
+DECLARE
+    error_msg text;
+    query1 text;
+    query2 text;
+BEGIN
+
+    query1 := pg_catalog.format('alter extension babelfishpg_common drop %s %s.%s', object_type, schema_name, object_name);
+    query2 := pg_catalog.format('drop %s %s.%s', object_type, schema_name, object_name);
+
+    execute query1;
+    execute query2;
+EXCEPTION
+    when object_not_in_prerequisite_state then --if 'alter extension' statement fails
+        GET STACKED DIAGNOSTICS error_msg = MESSAGE_TEXT;
+        raise warning '%', error_msg;
+    when dependent_objects_still_exist then --if 'drop view' statement fails
+        GET STACKED DIAGNOSTICS error_msg = MESSAGE_TEXT;
+        raise warning '%', error_msg;
+end
+$$
+LANGUAGE plpgsql;
+
+-- (sys.VARCHAR AS pg_catalog.TIME)
+DROP CAST IF EXISTS(sys.VARCHAR AS pg_catalog.TIME);
+
+DO $$
+DECLARE
+    exception_message text;
+BEGIN
+    ALTER FUNCTION sys.varchar2time(sys.VARCHAR) RENAME TO varchar2time_deprecated_4_4_0;
```

Comment:
We can directly remove the previous definition, we don't need to mark the function as deprecated because we have already dropped the dependent cast.

Reply:
Apparently the view definition does not replace the cast with cast function when it is stored.
Still just to be safe we could rename these sql functions.

Reply:
>Apparently the view definition does not replace the cast with cast function when it is stored.

yes, that is why we can directly drop and re-create it and we don't need to mark it as deprecated.

> Still just to be safe we could rename these sql functions.
We can have sufficient dependency upgrade test to validate that.

Reply:
The cast is dependent on the function (varchar2time) .Dropping the function directly results in failure of upgrade tests. 

**verification**:
1) created a view that used this cast.
`create view babel_5179_v1 as select cast(cast('12:45:37.123' as varchar) as time)`

2)Finding the oid of the view.
`select oid from pg_class where relname = 'babel_5179_v1';`

3)Finding the query using this oid
`select * from pg_rewrite where ev_class=19764;`

4)Finding the funcid from the query:
= 17090

5) Finding Which function is dependent on this cast:
`select proname from pg_proc where oid=17090;`

output: 
proname    
--------------
 varchar2time

**conclusion**: This view is dependent on varchar2time function. So we are deprecating this function instead of droppiing it directly.

cc: @Deepesh125

Reply:
Can you explain where the above gets used? If you mean the original views will still use the original definition then we should not add DEFAULT in sql definition suggested below, in that case we should check if the arg is present then only use it.

Reply:
Yes, that's the case where the original views will still use the original definition, also stated [above](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2877#discussion_r1733309961).

@Yvinayak07 please refer to the [suggestion](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2877#discussion_r1732485566).